app/main.py

```diff
@@ -1,13 +1,30 @@
-def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+def format_linter_error(error: dict[str, str]) -> dict[str, str]:
+    return {
+        "line": error.get("line_number"),
+        "column": error.get("column_number"),
+        "message": error.get("text"),
+        "name": error.get("code"),
+        "source": "flake8"
+    }
 
 
-def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+def format_single_linter_file(
+    file_path: str,
+    errors: list[dict[str, str]]
+) -> dict[str, any]:
+    return {
+        "errors": [
+            format_linter_error(error) for error in errors
+        ],
+        "path": file_path,
+        "status": "passed" if not errors else "failed"
+    }
 
 
-def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+def format_linter_report(
+    linter_report: dict[str, list[dict[str, str]]]
+) -> list[dict[str, any]]:
```

Comment:
"any" is a function
maybe u mean "Any" but it's incorrect typehint

add right typehint instead of "any"